src/parser/retail/modules/items/thewarwithin/trinkets/SpymastersWeb.tsx

```diff
@@ -0,0 +1,125 @@
+import ITEMS from 'common/ITEMS/thewarwithin/trinkets';
+import Analyzer, { Options } from 'parser/core/Analyzer';
+import Abilities from 'parser/core/modules/Abilities';
+import SPELLS from 'common/SPELLS/thewarwithin/trinkets';
+import SPELL_CATEGORY from 'parser/core/SPELL_CATEGORY';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import Events, { ApplyBuffEvent, ApplyBuffStackEvent, CastEvent } from 'parser/core/Events';
+import BoringItemValueText from 'parser/ui/BoringItemValueText';
+import { formatDuration, formatNumber } from 'common/format';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import { calculatePrimaryStat } from 'parser/core/stats';
+
+type SpymastersWebCast = {
+  timestamp: number;
+  stacks: number;
+};
+
+/**
+ * Based on the stats provided on wowhead.
+ *
+ * https://www.wowhead.com/item=220202/spymasters-web
+ */
+const SPYMASTERS_WEB_BASE_ILVL = 571;
+const SPYMASTERS_WEB_BASE_GAIN = 515;
+
+export default class SpymastersWeb extends Analyzer.withDependencies({
+  abilities: Abilities,
+}) {
+  protected currentReportStackCount: number = 0;
+  protected primaryStatBonus: number = 0;
+  protected SpymastersWebCasts: SpymastersWebCast[] = [];
+
+  constructor(options: Options) {
+    super(options);
+
+    // Ensure the combatant has the trinket.
+    this.active = this.selectedCombatant.hasTrinket(ITEMS.SPYMASTERS_WEB.id);
+    if (!this.active) {
+      return;
+    }
+
+    // Add the ability to the spellbook of the current combatant.
+    this.deps.abilities.add({
+      spell: SPELLS.SPYMASTERS_WEB.id,
+      category: SPELL_CATEGORY.ITEMS,
+      cooldown: 20,
+    });
+
+    this.addEventListener(Events.applybuff, this._onBuffApply);
+    this.addEventListener(Events.applybuffstack, this._onBuffApply);
```

Comment:
Is there a reason you don't do the spell filtering here instead of checking inside the functions?

Reply:
Good catch, totally forgot that htis was a thing...